docs/extend/api.md

```diff
@@ -0,0 +1,331 @@
+
+Also, remember that route names (`tags.index`, `tags.show`, etc) must be unique!
+:::
+
+The `Flarum\Api\Controller` namespace contains a number of abstract controller classes that you can extend to easily implement your JSON-API resources.
+
+### Listing Resources
+
+For the controller that lists your resource, extend the `Flarum\Api\Controller\AbstractListController` class. At a minimum, you need to specify the `$serializer` you want to use to serialize your models, and implement a `data` method to return a collection of models. The `data` method accepts the `Request` object and the tobscure/json-api `Document`.
+
+```php
+use Flarum\Api\Controller\AbstractListController;
+use Psr\Http\Message\ServerRequestInterface as Request;
+use Tobscure\JsonApi\Document;
+
+class ListTagsController extends AbstractListController
+{
+    public $serializer = TagSerializer::class;
+    
+    protected function data(Request $request, Document $document)
+    {
+        return Tag::all();
+    }
+}
+```
+
+#### Pagination
+
+You can allow the number of resources being **listed** to be customized by specifying the `limit` and `maxLimit` properties on your controller:
+
+```php
+    // The number of records included by default.
+    public $limit = 20;
+    
+    // The maximum number of records that can be requested.
+    public $maxLimit = 50;
+```
+
+You can then extract pagination information from the request using the `extractLimit` and `extractOffset` methods:
+
+```php
+$limit = $this->extractLimit($request);
+$offset = $this->extractOffset($request);
+
+return Tag::skip($offset)->take($limit);
+```
+
+To add pagination links to the JSON:API document, use the `Document::addPaginationLinks` method.
+
+#### Sorting
+
+You can allow the sort order of resources being **listed** to be customized by specifying the `sort` and `sortField` properties on your controller:
+
+```php
+    // The default sort field and order to use.
+    public $sort = ['name' => 'asc'];
+    
+    // The fields that are available to be sorted by.
+    public $sortFields = ['firstName', 'lastName'];
+```
+
+You can then extract sorting information from the request using the `extractSort` method. This will return an array of sort criteria which you can apply to your query:
+
+```php
+$sort = $this->extractSort($request);
+$query = Tag::query();
+
+foreach ($sort as $field => $order) {
+    $query->orderBy(snake_case($field), $order);
+}
+
+return $query->get();
+```
+
+#### Searching and Filtering
+
+Read our [searching and filtering](search.md) guide for more information!
+
+### Showing a Resource
+
+For the controller that shows a single resource, extend the `Flarum\Api\Controller\AbstractShowController` class. Like for the list controller, you need to specify the `$serializer` you want to use to serialize your models, and implement a `data` method to return a single model.
+We'll learn about serializers [in just a bit](#serializers). 
+
+
```

Comment:
Is there meant to be a new paragraph between 130 and 131? We can also remove one of the line breaks on 132/133